### PR TITLE
feat: content grid

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

This PR introduces a new class `.content-grid` that can be use to present the content in a flex / grid view for small / wider devices. The grid also support up to two columns and rows.

PR also renames previous `.grid` global class to `.list-grid`.

# Changes

- lib: rename `.grid` to `.card-grid`
- lib: introduces `.content-grid`
- docs: improve documentation of the grids style
- docs: refactor and split "styling" related information in dedicated pages

# Screenshots Docs

Capture d’écran 2022-08-23 à 10 02 14
Capture d’écran 2022-08-23 à 10 02 17
Capture d’écran 2022-08-23 à 10 02 22
Capture d’écran 2022-08-23 à 10 02 26
Capture d’écran 2022-08-23 à 10 02 33

### CODE REVIEW

File: src/lib/styles/global/grid.scss [hunk 1]
```diff
@@ -20,3 +20,39 @@
...
+    column-gap: var(--column-gap);
+    min-height: auto;
+
+    grid-template-rows: auto;
+    grid-template-areas:
+      "item-a item-a item-a item-a item-a item-a item-b item-b item-b item-b item-b item-b"
+      "item-c item-c item-c item-c item-c item-c item-d item-d item-d item-d item-d item-d";
+  }
+
+  .item-a {
```
Comment (mstrasinskis) [inline src/lib/styles/global/grid.scss:43]: Just because of the `.content-grid`: `.item-a` -> `.content-a`?

**Review by mstrasinskis — APPROVED** [review]
LGTM. Just a couple of minor comments.

File: src/docs/styles/docs-global.scss [hunk 2]
```diff
@@ -40,7 +40,7 @@ main {
   }
 
   & > h2,
-  & > .grid {
+  & > .list-grid {
```
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply src/docs/styles/docs-global.scss]: > Not part of this PR but is having this part & > necessary?

If I remember correctly yes but, it's specific to the documentation. `docs-global.scss` is not a CSS that is exported in the library but used only for the showcase.

> Name suggestion: card-grid (no a strong opinion 😉 )

Like it 👍. Done.

File: src/lib/styles/global/grid.scss (hunk 1 above)
Comment (peterpeterparker) [reply src/lib/styles/global/grid.scss]: Like it too! Done.

**Pull request merged**